### PR TITLE
Get rid of VectorSpaceVector in the LinearAlgebra::distributed::BlockVector class.

### PR DESCRIPTION
**Pull request by bangerth**
Step 4 for #3073.

### CODE REVIEW

File: include/deal.II/lac/la_parallel_block_vector.h [hunk 1]
```diff
@@ -513,41 +512,33 @@ namespace LinearAlgebra
       /** @} */
 
       /**
-       * @name 2: Implementation of VectorSpaceVector
+       * @name 2: Implementation of vector space vector interface
```
Comment (masterleinad) [inline include/deal.II/lac/la_parallel_block_vector.h:515]:
```suggestion
       * @name 2: Implementation of the vector space vector interface
```
?

**Review by masterleinad — APPROVED** [review]

File: include/deal.II/lac/la_parallel_block_vector.h (hunk 1 above)
Comment (bangerth) [reply include/deal.II/lac/la_parallel_block_vector.h]: I named it completely differently.

**Review by masterleinad — APPROVED** [review]

**Comment by tamiko** [comment]
We have clang tidy complaining with:
```
/home/runner/work/dealii/dealii/include/deal.II/lac/la_parallel_block_vector.templates.h:863:70: error: the parameter #3 is copied for each invocation but only used as a const reference; consider making it a const reference [performance-unnecessary-value-param,-warnings-as-errors]
```

**Review by kronbichler — APPROVED** [review]

**Pull request merged**